required/latex-lab/latex-lab-math.dtx

```diff
@@ -2568,6 +2437,257 @@
+    {
+      \tl_gset:Nn \g_@@_skip_sign_tl {-}
+      \skip_set:Nn \belowdisplayskip      {-\belowdisplayskip}
+      \skip_set:Nn \belowdisplayshortskip {-\belowdisplayshortskip}
+    }
+%    \end{macrocode}
+%    For the skip it is enough to flip the sign, because we get the
+%    value back through \cs{lastskip} but for the
+%    \cs{postdisplaypenalty} change we have to  save the current value
+%    somewhere, because it may have been changed withing the display
+%    formula. This has to be done globally, because it is used after
+%    the formula has ended.
+%    \begin{macrocode}
+  \int_gset_eq:NN \g_@@_postdisplaypenalty_int \postdisplaypenalty
+  \int_set_eq:NN  \postdisplaypenalty \@M
+}
+%    \end{macrocode}
+% \end{macro}
+%    
+%    
+%
+% \begin{macro}{\@@_tag_dollardollar_display_end:}
+%
+%    The code to be executed after the formula has ended is added
+%    through a \cs{group_insert_after:N} in \cs{tex_everydisplay:D},
+%    i.e., when the formula starts. Thus, it has to be stored in a
+%    command
+% \changes{v0.6j}{2024-11-19}{removed unneeded \cs{tagpdfparaOn}}
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_tag_dollardollar_display_end:
+  {
+    %  \typeout{== tag dollarldollar display end}
+    %  \ShowTagging{struct-stack}
+    \para_raw_end:
+%    \end{macrocode}
+%    The \cs{postdisplaypenalty} was temporarily set
+%    to 10000 inside the display and the \cs{belowdisplayskip} and the
+%    \cs{belowdisplayshortskip} was negated (with some exceptions, see
+%    above), so whatever was inserted
+%    it should have been a negative skip. Whatever
+%    was added, we pick up the value, so that we can correct
+%    the spacing after the tagging code has been inserted.
+%    \begin{macrocode}
+    \l_@@_tmpa_skip \lastskip
+    \tag_socket_use:n{math/display/formula/end}
+%    \end{macrocode}
+%    Now we add a skip without introducing a page break possibility,
+%    that should bring the current vertical position back to the point
+%    where \TeX{} would add the penalty and the \enquote{below skip}.
+% \changes{v0.6f}{2024-09-30}{Correct logic for inserting below skips
+%                             after displays (tagging/721)}
+%    \begin{macrocode}
+    \nobreak
+    \skip_vertical:n { -\l_@@_tmpa_skip  }
+%    \end{macrocode}
+%    Then we finally add the real stuff: the true \cs{postdisplaypenalty}
+%    (saved in \cs{g_@@_postdisplaypenalty_int} earlier)
+%    and then negated value of skip we saved from above. It may look
+%    strange that we have two identical negated skips next to each
+%    other, but if you think about it, that is correct: the first
+%    cancels the \enquote{below skip} that \TeX{} had added and the
+%    second puts the same amount after the penalty (which is where it
+%    should be).
+%    \begin{macrocode}
+    \penalty \g_@@_postdisplaypenalty_int
+    \skip_vertical:n
+%    \end{macrocode}
+%    Actually we may just use the skip withought flipping the sign
```

Comment:
without

Reply:
within and without are both up to no good!